client/landing/stepper/declarative-flow/internals/index.tsx

```diff
@@ -159,8 +159,8 @@ export const FlowRenderer: React.FC< { flow: Flow } > = ( { flow } ) => {
 	}, [ location ] );
 
 	useEffect( () => {
-		if ( isFlowStart() ) {
-			recordSignupStart( flow.name, ref );
+		if ( flow.isSignupFlow && isFlowStart() ) {
+			recordSignupStart( flow.name, ref, flow.useSignupStartEventProps?.() ?? {} );
```

Comment:
Would be better to update the recordSignupStart to have the third param with a default value? 
`recordSignupStart(param1, param2, param3={})`.

The effect is the same but it will make the code describing our usage better.

Reply:
When `useSignupStartEventProps` is defined, Stepper breaks with the message: 

`Uncaught Error: Invalid hook call. Hooks can only be called inside of the body of a function component.`

Reply:
Just noting here as well that I'm removing the `flow.useSignupSstartEventProps` call to unblock this PR. We can iterate on it.